pandas/tests/extension/arrow/test_timestamp.py

```diff
@@ -0,0 +1,20 @@
+import datetime
+
+import pytest
+
+import pandas as pd
+
+pytest.importorskip("pyarrow", minversion="0.13.0")
+
+from .arrays import ArrowTimestampUSArray  # isort:skip
+
+
+def test_constructor_extensionblock():
```

Comment:
should be xfailed?

Reply:
> I would prefer to fix this myself though.

Sounds good.  Is this a use case you have a need to get working near-term, or more of a Principle Of The Thing?  I ask because...

> I just need a pointer at which code section I should apply a fix.

This is pretty daunting, as I expect this is scattered across the code.  There are lots of places where we either a) implicitly assume nanoseconds or b) check `dtype.kind in ["M", "m"]` (much more performant than the `is_foo_dtype` checks)

> Should I change the order in pandas/pandas/core/internals/blocks.py so that we only create a DatetimeTZBlock for pandas-provided datetime-based ExtensionArrays

That will probably be part of a solution.

> or shouldn't is_datetime64tz_dtype return True for my ExtensionDtype?

I'd be very reticent to make that change, since I think a lot of code expects that to imply its getting our Datetime64TZDtype.  Maybe a `is_3rd_party_ea_dtype` that we would check for before checking for any 1st-party dtypes?  That runs into the "ideally we should treat 3rd party EAs symmetrically with 1st-party" problems.

So getting back to the motivation: how high a priority is this?

One thing I can unambiguously encourage is more tests, even if xfailed: 
- what happens if you pass one of these to the DatetimeIndex constructor?  vice-versa?
- what happens if i do DatetimeIndex.astype(this_new_ea_dtype)
- addition/subtraction with the gamut of datetime/timedelta scalars/arrays we already support?
- How does this behave if you stuff it inside a Categorical/CategoricalIndex?

Reply:
> > I would prefer to fix this myself though.
> 
> Sounds good. Is this a use case you have a need to get working near-term, or more of a Principle Of The Thing? I ask because...

More in the next 6 months range, thus I'm definitely going to add an `xfail` here as the points below indicate that we should rather think more than "fix quick".

I would love to have a nullable, non-nanosecond timestamp (actually I desparately need it but e.g. having a performant string is more important to me) but there are several other places that either assume that all timestamps are nanoseconds or backed by a numpy-array, so this is going to be a major effort. 

> > or shouldn't is_datetime64tz_dtype return True for my ExtensionDtype?
> 
> I'd be very reticent to make that change, since I think a lot of code expects that to imply its getting our Datetime64TZDtype. Maybe a `is_3rd_party_ea_dtype` that we would check for before checking for any 1st-party dtypes? That runs into the "ideally we should treat 3rd party EAs symmetrically with 1st-party" problems.
> 
> So getting back to the motivation: how high a priority is this?

As already pointed out: Less than other things I want to contribute to `pandas`, so `xfail`ing and adding more (possibly) xfailing tests is the way to go.

Reply:
> so xfailing and adding more (possibly) xfailing tests is the way to go.

Sounds good.

> actually I desparately need  [...] that either assume that all timestamps are nanoseconds or backed by a numpy-array

Would your need be solved if we get numpy-backed non-nano in place?  There's a reasonable chance of that happening in the next 6 months.

Reply:
> Would your need be solved if we get numpy-backed non-nano in place? There's a reasonable chance of that happening in the next 6 months.

For now: Yes.

Reply:
> For now: Yes.

I'm slowly tackling this from the cython side of the code.  The parallelizable step is to comb through the rest of the code to find all the places where we implicitly/explicitly assume nanos.  I'd start with pandas/plotting and pandas/io.

Reply:
lets see if we can at least get this one working.

i think we'll need to edit the dtype.kind check in is_datetime64tz_dtype, and possible the `issubclass(vtype, np.datetime64)` check in internals.blocks.get_block_type